tests/scenarios/v2-addon-dev-test.ts

```diff
@@ -0,0 +1,209 @@
+        'package.json': JSON.stringify(packageJSON, null, 2),
+        config: {
+          'addon-main.cjs': `
+            const path = require('path');
+            const { addonV1Shim } = require('@embroider/addon-shim');
+
+            module.exports = addonV1Shim(path.dirname(__dirname));
+          `,
+        },
+        'babel.config.json': `
+          {
+           "presets": [
+              ["@babel/preset-env", {
+                "targets": ["last 1 firefox versions"]
+              }]
+            ],
+            "plugins": [
+              "@embroider/addon-dev/template-colocation-plugin",
+              ["@babel/plugin-proposal-decorators", { "legacy": true }],
+              [ "@babel/plugin-proposal-class-properties" ]
+            ]
+          }
+        `,
+        'rollup.config.js': `
+          import babel from '@rollup/plugin-babel';
+          import { Addon } from '@embroider/addon-dev/rollup';
+
+          const addon = new Addon({
+            srcDir: 'src',
+            destDir: 'dist',
+          });
+
+          export default {
+            output: addon.output(),
+
+            plugins: [
+              addon.publicEntrypoints([
+                '**/*.js',
+                'components/demo/out.hbs',
+              ]),
+
+              addon.appReexports([
+                'components/**/*.js',
+                'components/demo/out.hbs',
+              ]),
+
+              addon.hbs(),
+              addon.dependencies(),
+
+              babel({ babelHelpers: 'bundled' }),
+
+              addon.clean(),
+            ],
+          };
+        `,
+        components: {
+          demo: {
+            'button.hbs': `
+              <button {{on 'click' @onClick}}>
+                flip
+              </button>
+            `,
+            'out.hbs': `
+              <out>{{yield}}</out>
+            `,
+            'index.js': `
+                import Component from '@glimmer/component';
+
+                import FlipButton from './button';
+                import BlahButton from './button.hbs';
+                import Out from './out';
+
+                export default class ExampleComponent extends Component {
+                  Button = FlipButton;
+                  Button2 = BlahButton;
+                  Out = Out;
+
+                  @tracked active = false;
+
+                  flip = () => (this.active = !this.active);
+                }
+              `,
+            'index.hbs': `
+              Hello there!
+
+              <this.Out>{{this.active}}</this.Out>
+
+              <this.Button @onClick={{this.flip}} />
+              <this.Button2 @onClick={{this.flip}} />
+            `,
+          },
+        },
+      },
+    });
+
+    addon.linkDependency('@embroider/addon-shim', { baseDir: __dirname });
+    addon.linkDependency('@embroider/addon-dev', { baseDir: __dirname });
+
+    await addon.write();
+    let addonDir = addon.baseDir;
+
+    await execa('yarn', ['install'], { cwd: addonDir });
```

Comment:
Most definitely do not do this. The entire reason scenario-tester exists is to avoid this. This is why we have `project.linkDependency` and `Project.fromDir(whatever, { linkDeps: true })`.

I realize there's no teaching materials about this, but as a general point of advice: to learn how a system like this works, read any other test. All of them need dependencies too, and none of them call yarn install to get them.

Reply:
I learned that the node_modules are nested within the test app's node_modules, and it meant that `yarn` also couldn't be used to start rollup -- had to reach in to the package and run the bin directly